sdk/src/nonce_instruction.rs

```diff
@@ -63,12 +63,25 @@ pub enum NonceInstruction {
     ///     0 - A NonceAccount in the Uninitialized state
     ///     1 - RecentBlockHashes sysvar
     ///     2 - Rent sysvar
-    Initialize,
+    ///
+    /// The `Pubkey` parameter specifies the entity authorized to execute nonce
+    /// instruction on the account
+    Initialize(Pubkey),
```

Comment:
Passing Pubkeys in as instruction data is clunky, it requires custom serde code.  Instead just add the pubkey as one of the required non-signer keys

Reply:
Saves having to pull that account, clone its data, etc...

Reply:
Left this one for the time being.  Seems like the runtime savings should take precedent?  wdyt @mvines ?

Reply:
Ok fine.  You win this round. 🤼‍♂